### PR TITLE
Process 176 setup branching policy

### PR DESCRIPTION
**Pull request by hellwolf**

### CODE REVIEW

**Comment by hellwolf** [comment]
@pi0neerpat good stuff, I did bunch of experiment, let's sync in one go instead of typing here one by one.

**Comment by hellwolf** [comment]
@pi0neerpat first of all, it seems you would still need to use github token to access the github registry:

for instance I would set globally like:

```
master $ cat ~/.npmrc 
//prefix=/home/hellwolf/.npm-packages
registry=https://registry.npmjs.com/
//npm.pkg.github.com/:_authToken=${GITHUB_TOKEN}
//registry.npmjs.com/:_authToken=${NPMJS_TOKEN}
```

After that, I managed to  do the `yarn add` command as the github action bot nicely put a message there.

Also, I don't know what yarn's equivalent command is, I can see this with npm command:

```
$ npm --registry https://npm.pkg.github.com/ show @superfluid-finance/js-sdk

@superfluid-finance/js-sdk@0.2.2 | MIT | deps: 1 | versions: 15
Javascript SDK for building with Superfluid Protocol
https://github.com/superfluid-finance/protocol-monorepo#readme

dist
.tarball: https://npm.pkg.github.com/download/@superfluid-finance/js-sdk/0.2.2/a3c061eb89808ca85983d0439ecf67acbb2340fcb8108911ca6c6f729604ae69
.shasum: 27c85514ceb7b0d0649c3a8846e1ff0275aff72c
.integrity: sha512-rrdBDJ95bn84axdMmCUP3mWUy7JuEt5pIVRkR2L6izqcqe4rEIk2Ej4Tojg+ulAEUnn7sJPahfHbsTBk1Ase2A==

dependencies:
@truffle/contract: 4.3.4 

dist-tags:
196.461d3570b1fbc68e21d809a6b0b83473d8c3b26c: 0.2.3-PR196.461d3570b1fbc68e21d809a6b0b83473d8c3b26c.0
196.a4ac5dbf78107d02d7d3bbf4d189ff18ad66f2c5: 0.2.3-PR196.a4ac5dbf78107d02d7d3bbf4d189ff18ad66f2c5.0
196.df11e427a685def8b22368ab6a284daf92beaffa: 0.2.3-PR196.df11e427a685def8b22368ab6a284daf92beaffa.0
196.f6caea75a3ac406476e38bf884eafc0d20311768: 0.2.3-PR196.f6caea75a3ac406476e38bf884eafc0d20311768.0
196: 0.2.3-PR196.0
PR196.4efe3f1: 0.2.3-PR196.4efe3f1.0
latest: 0.2.2

published 5 days ago by undefined <undefined>
```

**Comment by hellwolf** [comment]
publish both to npmjs and github registry for canary packages, due to the inability to use both registry in any private superfluid repo.

**Comment by pi0neerpat** [comment]
@hellwolf  ready for review. Upon merging we should see a new canary release :crossed_fingers: on github + npm.

I will get my signed-commits set up soon!

File: .github/workflows/ci-canary.yml [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+          git fetch --depth=1 origin +refs/tags/*:refs/tags/*
+          echo "//registry.npmjs.org/:_authToken=${NPMJS_TOKEN}" > .npmrc
+          ./tasks/publish-canary.sh
+        env:
+          NPMJS_TOKEN: ${{ secrets.NPMJS_TOKEN }}
+
+  publish:
+    name: NPM publish dev branch
+
+    if: github.repository == 'superfluid-finance/protocol-monorepo'
```
Comment (hellwolf) [inline .github/workflows/ci-canary.yml:55]: btw, did you do experiment with PR from a forked repository?

File: .github/workflows/ci-canary.yml [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+
+      - name: Build
+        run: yarn build
+
+      - name: Publish to npm
+        run: |
+          git fetch --depth=1 origin +refs/tags/*:refs/tags/*
```
Comment (hellwolf) [inline .github/workflows/ci-canary.yml:76]: is this necessary? I removed this in test-pr.yml

File: .github/workflows/ci-canary.yml [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+          echo "//registry.npmjs.org/:_authToken=${NPMJS_TOKEN}" > .npmrc
+          ./tasks/publish-canary.sh
+        env:
+          NPMJS_TOKEN: ${{ secrets.NPMJS_TOKEN }}
+
+  publish:
```
Comment (hellwolf) [inline .github/workflows/ci-canary.yml:52]: do we publish even when build failed?

File: .github/workflows/ci-canary.yml [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+
+      - name: Create coverage artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: coverage
+          path: |
+            packages/ethereum-contracts/coverage/
+            packages/ethereum-contracts/coverage.json
+
+  deploy-dev:
```
Comment (hellwolf) [inline .github/workflows/ci-canary.yml:119]: should check  github.repository too

File: .github/workflows/test-pr.yml [hunk 5]
```diff
@@ -1,4 +1,4 @@
-name: Test branch code
+name: Test Pull Request
```
Comment (hellwolf) [inline .github/workflows/test-pr.yml:1]: to be precise though, it to test anything pushed to branches other than "master" or "dev"

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
+# Contributing
```
Comment (hellwolf) [inline CONTRIBUTING.md:1]: Good stuff!

**Comment by hellwolf** [comment]
@pi0neerpat please approve this PR, since it's me created it.

I don't see publish-pr being triggered, I guess it has to be in the default branch (`dev` in this case) in the first place?

**Review by pi0neerpat — APPROVED** [review]
:+1:

**Pull request merged**